### DISCUSSION TITLE
Email inbox as knowledge

### DISCUSSION THREAD

**Discussion by chrigui** [langgenius/dify#3948] [General]
how would you approach using gmail email inbox as a knowledge for a chat bot ?

Back story:
we use email to manage multiple projects over emails, Like talking to clients, suppliers, compliance and internal teams.
each project has its own ID and its own team.

Goal:
Be able to chat with Dify assistant to request information about project progress, information, status of certain tasks.

is it possible ?
How would you do it if its doable ?

Many thanks

**Comment by zswll2** [comment]
Great idea, I also need this feature. I believe that in order to support this functionality, dify must support the functionality of an SMTP server.

**Reply by chrigui** [reply]
yes i think the first step is to support SMTP, the largest challenge would be the live search of inbox...

**Reply by zswll2** [reply]
My idea is to import all emails through the knowledge base API and classify them according to specific requirements, so that I can search for the required emails and reply or query them.

The difficulty lies in complete automation, which requires the development of a completely new module to implement SMTP email collection, import knowledge base according to requirements, query and reply operations, etc

I also really need this feature, I think it can improve a lot of efficiency for enterprises, but it should require a lot of development support I hope it can be achieved in the future
